Scripts/Services/Harvest/Fishing/Fishing.cs

```diff
@@ -771,7 +771,15 @@ public override bool BeginHarvesting(Mobile from, Item tool)
             if (!base.BeginHarvesting(from, tool))
                 return false;
 
-            from.SendLocalizedMessage(500974); // What water do you want to fish in?
+            if (tool is FishingPole pole && pole.HookType == HookType.Lava)
```

Comment:
The tool should be fishing pole in both cases.

Reply:
Please check the newest changes @Argalep